CMakeLists.txt

```diff
@@ -422,10 +422,34 @@ if (WABT_INSTALL_RULES)
   )
 endif ()
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads)
+
+add_custom_command(
+  OUTPUT gen-wasm2c-rt-prebuilt
+
+  COMMAND ${CMAKE_COMMAND} -D out="${WABT_SOURCE_DIR}/wasm2c/prebuilt/wasm-rt-mem-impl.c" -D in="${WABT_SOURCE_DIR}/wasm2c/template/wasm-rt-mem-impl.tmpl.c" -D replace="" -P ${WABT_SOURCE_DIR}/scripts/gen-wasm2c-rt-templates.cmake
+  COMMAND ${CMAKE_COMMAND} -D out="${WABT_SOURCE_DIR}/wasm2c/prebuilt/wasm-rt-mem-shared-impl.c" -D in="${WABT_SOURCE_DIR}/wasm2c/template/wasm-rt-mem-impl.tmpl.c" -D replace="_shared" -P ${WABT_SOURCE_DIR}/scripts/gen-wasm2c-rt-templates.cmake
```

Comment:
In general, the typical reasons for checking in the prebuilt file are
1. We can see the output which helps code reviews
2. Someone who builds the code doesn't have to install dependencies that do code generation

In wasm2c's use cases however, point 1 isn't really useful as the template files are more readable than the output; and point 2 only uses cmake and python which are both already required for the build. I am not opposed to making this part of the build directory, but this is something we can do in a separate PR if needed since it has to modify other template builds unrelated to this change.

Reply:
I think (1) applies to the other prebuilts, but perhaps not so much to this one (where input and output look basically identical).    So if we do go with templates here (which I would prefer that we simply avoid) then maybe we don't need to check in the prebuilts in this case.

(BTW I don't think python is hard dependency at build time, is it?)

Reply:
Not sure. There is this line in the CMake
https://github.com/WebAssembly/wabt/blob/main/CMakeLists.txt#L666

But I don't know what the "COMPONENTS Interpreter" part does... From context, it would seem python is required for the build if you build the interpreter in wabt. 

If it isn't a hard dependency at the moment, any build time codegen (vs. the current checkin time codegen) will make python a hard build dependency

Reply:
But we don't use python to generate any of the codegen stuff do we?   IIUC its just used for testing.   (Sorry we might be getting off topic here).

Reply:
oh my mistake, we don't use python for this template generation. This only requires CMake. Given this, would you prefer to change this to generate in the build folder?

Reply:
also @keithw thoughts on this?

Reply:
fixed